app/models/participant.rb

```diff
@@ -168,4 +168,8 @@ def get_scores(questions)
 
       scores
     end
+
+    def self.find_participant_byassign(user_id,assign_id)
```

Comment:
We rewrite this function according to the original code "participant =
Participant.where(user_id: session[:user].id, parent_id:
assignment_id).first".
We did not check if the original code miss some important checks. I think
for this part, our task is just  to improve format and style.

On Thu, Apr 16, 2015 at 3:03 PM, Yang Song <notifications@github.com> wrote:

> In app/models/participant.rb
> <https://github.com/expertiza/expertiza/pull/531#discussion_r28539561>:
>
> > @@ -168,4 +168,8 @@ def get_scores(questions)
> >
> >        scores
> >      end
> > +
> > +    def self.find_participant_byassign(user_id,assign_id)
>
> You should also check if the participant.type is "AssignmentParticipant".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/expertiza/expertiza/pull/531/files#r28539561>.
>